nullaway/src/main/java/com/uber/nullaway/NullAway.java

```diff
@@ -779,7 +779,7 @@ private boolean fieldAlwaysInitializedBeforeRead(
     Tree readExprTree = pathToRead.getLeaf();
     int readStartPos = getStartPos((JCTree) readExprTree);
     TreePath classTreePath = enclosingBlockPath;
-    while (!classTreePath.getLeaf().getKind().equals(Tree.Kind.CLASS)) {
+    while (!(classTreePath.getLeaf() instanceof ClassTree)) {
```

Comment:
Wait, is the error here that `classTreePath.getLeaf()` is not `null` but `classTreePath.getLeaf().getKind()` returns `null`? That's weird, `getKind()` was supposedly preferred vs `instanceof` for `Tree`s, was it not?

Reply:
How about `Tree.Kind.CLASS.equals(classTreePath.getLeaf().getKind())` or is it too likely people will again switch that around?

Reply:
Or, wait, is `getLeaf()` `null`? Because then maybe splitting it into an explicit null check and the `getKind()` stuff is better...

Reply:
Sorry, should have explained better.  The issue is that trees for enum types do not have `CLASS` kind but can contain initialization code.  Since we were only checking for `CLASS` kind before, this parent traversal wouldn't find anything, eventually ending up in a state where `classTreePath` is `null`, leading to an NPE.  What we really care about is finding the enclosing `ClassTree` (which can represent a class, enum, or interface), so this checks that more directly.  Will add a code comment.

Reply:
Oh, I see. So checking by `Kind` would require three separate checks and might still break if something else that subclasses `ClassTree` gets added. Makes sense now.